src/components/text/text.styles.ts

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { css } from '@emotion/react';
+import { UseEuiTheme } from '../../services';
+import {
+  logicalCSS,
+  euiFontSize,
+  _FontScaleOptions,
+} from '../../global_styling';
+
+import { euiLinkCSS } from '../link/link.styles';
+import { euiTitle } from '../title/title.styles';
+
+export const euiTextConstrainedMaxWidth = '36em';
+
+/**
+ * Mixins
+ */
+export const euiText = (
+  euiTheme: UseEuiTheme['euiTheme'],
+  inheritColor = false
+) => {
+  return {
+    color: inheritColor ? 'inherit' : euiTheme.colors.text,
+    fontWeight: euiTheme.font.weight.regular,
+  };
+};
+
+// Internal utility for EuiText scales/sizes
+const euiScaleText = (
+  euiTheme: UseEuiTheme['euiTheme'],
+  options: _FontScaleOptions
+) => {
+  const { fontSize, lineHeight } = euiFontSize('m', euiTheme, options);
+  const { measurement } = options;
+  const lineHeightSize = measurement === 'em' ? `${lineHeight}em` : lineHeight;
+
+  const headings = {
+    h1: euiTitle('l', euiTheme, options),
+    h2: euiTitle('m', euiTheme, options),
+    h3: euiTitle('s', euiTheme, options),
+    h4: euiTitle('xs', euiTheme, options),
+    h5: euiTitle('xxs', euiTheme, options),
+    h6: euiTitle('xxxs', euiTheme, options),
+  };
+  const headingMarginTop = (
+    headingFontSize: ReturnType<typeof euiTitle>['fontSize']
+  ) => {
+    let marginTop = parseFloat(String(fontSize)) * 2;
+    if (measurement === 'em') {
+      marginTop = marginTop / parseFloat(String(headingFontSize));
+    }
+    return `${marginTop}${measurement}`;
+  };
+  const headingMarginBottom = (
+    headingFontSize: ReturnType<typeof euiTitle>['fontSize']
+  ) => {
+    let marginBottom = parseFloat(String(fontSize)) / 2;
+    if (measurement === 'em') {
+      marginBottom = marginBottom / parseFloat(String(headingFontSize));
+    }
+    return `${marginBottom}${measurement}`;
+  };
+
+  return `
+    font-size: ${fontSize};
+    line-height: ${lineHeight};
+
+    h1 {
+      font-size: ${headings.h1.fontSize};
+      line-height: ${headings.h1.lineHeight};
+    }
+    h1:not(:last-child) {
+      ${logicalCSS('margin-bottom', headingMarginBottom(headings.h1.fontSize))}
+    }
```

Comment:
Looks like you're already doing that 😄 . I would be even more selective though and change it to `type` on all of them
```suggestion
    h1:not(:last-of-type) {
      ${logicalCSS('margin-bottom', headingMarginBottom(headings.h1.fontSize))}
    }
```

Reply:
Sorry, I'm not clear on the intention of this change this change. This will make it so something like

```
<EuiText>
  <h1>Hello</h1>
  <p>World</p>
</EuiText>
```
With `last-of-type` instead of `last-child`, the `h1` title **will not** have a a margin-bottom where previously it would have a margin between it and the following `p` tag. I'm not clear on why that would be and if that's your intention with this selector.

Reply:
Oi.. brain no work good. yes you are correct, leave as `last-child`.